src/coprocessor/endpoint.rs

```diff
@@ -252,13 +133,10 @@ impl Host {
         let mut tracker = t.tracker;
         let ranges = req.take_ranges().into_vec();
 
-        let pool_and_ctx_pool = match req.get_context().get_priority() {
-            CommandPri::Low => &mut self.low_priority_pool,
-            CommandPri::High => &mut self.high_priority_pool,
-            CommandPri::Normal => &mut self.pool,
-        };
-        let pool = &pool_and_ctx_pool.pool;
-        tracker.ctx_pool(pool_and_ctx_pool.contexts.clone());
+        let priority = readpool::Priority::from(req.get_context().get_priority());
+        let pool = self.pool.get_pool_by_priority(priority);
+        let ctxd = pool.get_context_delegators();
+        tracker.ctx_pool(&ctxd);
```

Comment:
I think it's a bug. If we return at line 147, the `tracker` will be deconstructed out of pool. And I prefer clone here instead of in `fn ctx_pool`.

Reply:
Nice catch. I think this bug exists even before this PR. Since Coprocessor EndpointHost is totally rewritten in other PRs and there will be no such bug, I think it will be acceptable to not to fix it here.